src/plotViewer/index.ts

```diff
@@ -763,7 +763,29 @@ export class HttpgdViewer implements IHttpgdViewer {
         }
         // make sure outFile is valid or return:
         if (!outFile) {
+            // Suggest a file extension:
+            const renderer = this.api.getRenderers().find(r => r.id === rendererId);
+            
+            const defaultUri = vscode.workspace.workspaceFolders[0]?.uri;
```

Comment:
We might need to keep in mind that we support multi-root workspace. So it would be best if we could actually use a proper workspace folder that makes sense here. Need to cover the case of no workspace folder, single workspace folder, multi-root workspace.

Maybe we should use the workspace folder of the active document?

Reply:
Since the export dialogue will usually be triggered while the plot webview is active, the active document is `undefined` when this code runs.

Alternatively, we could start with the current `defaultUri`, but remember the save location and use it as default the next time the save dialogue is opened?

Reply:
Each plot viewer is started from an interactive R session. I don't know if each plot viewer remembers the working directory of that session.

Maybe we should use the working directory of the R session that starts the plot viewer as the default folder?

If the plot viewer do not know it then we might use the working directory of the currently attached session?

Reply:
> Each plot viewer is started from an interactive R session. 

An external httpgd-url can also be started using the commadn `r.plot.openUrl`. Using the wd of the connected R process might be the cleanest solution here. @nx10 would you consider adding the wd to the httpgd api or are there problems with this approach?

Reply:
Do you mean adding an API that returns the working directory path of the R session? I am not sure if this is the best approach as the server runs not necessarily on the same system as the client (remote connections). 
I just added the `/info` / `hgd_info()` API to both httpgd and httpgd-js, which reports version information (which may be used for feature detection) and a (universally) unique server/device ID. Could the ID be used to work around this problem?

Reply:
> Do you mean adding an API that returns the working directory path of the R session?

Yes. But I get the concern that the R session might be running behind a remote connection, and that it wouldn't make sense to reveal a file path in that case.

But then I would rather not try to guess a directory based on the server/device ID or wd of the attached R process (this might get messy, and choosing one of the workspace folders is at least a very predictable behaviour). Remembering the last used export path also feels quite natural as this is done in many other programs as well.

Reply:
One question is that downloading the plot will put the file in the client's local filesystem right? Under remote development, the workspace folder is still remote rather than local.

Reply:
> One question is that downloading the plot will put the file in the client's local filesystem right?

Not sure what you mean here. Exporting the plot will save it to the remote file system, as should be the expected behavior.

Reply:
I thought "Export to PNG/SVG/..." always saves the image to the client file system, at least it does in the browser.

I would be totally fine to start with the first workspace folder and remember user choice.